### PR TITLE
feat: added the UI for Home Page.

### PR DESCRIPTION
**Pull request by arteevraina**
- [x] Added the Home page UI.
- [x] Enable searching from Home Page.
- [x] Improved UI of Navbar Search.
- [x] Hide navbar search when the current route is Home Page.

### CODE REVIEW

**Comment by arteevraina** [comment]
@minhqdao https://registry-frontend.vercel.app/

**Comment by minhqdao** [comment]
The enter key now works on the homepage, but once you searched sth and the search bar moves to the top, the enter key stops working.

**Comment by minhqdao** [comment]
When I search for `asdf`, then go the homepage again via the `fpm` icon and search for `fpm`, then the old results are shown just quickly. It's not super important but I think the state should be resetted before the navigation (actually it should be coupled with the loading indicator screen).

File: registry/src/pages/home.js [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
+  const dispatch = useDispatch();
+
+  const search = () => {
+    if (query.trim().length !== 0) {
+      dispatch(searchPackage(query, 0));
+      navigate("/search");
+    }
+  };
+
+  const handleKeyDown = (event) => {
```
Comment (henilp105) [inline registry/src/pages/home.js:36]: This function must be embedded in both the home and the Navbar, for the search to work by enter Key.

**Comment by henilp105** [comment]
Thanks for the suggestions , @minhqdao I have fixed them in my fork , will be fixed in upcoming PR.

**Comment by henilp105** [comment]
Thanks @arteevraina , just a few suggestions above rest Looks good to me.

**Comment by arteevraina** [comment]
> When I search for `asdf`, then go the homepage again via the `fpm` icon and search for `fpm`, then the old results are shown just quickly. It's not super important but I think the state should be resetted before the navigation (actually it should be coupled with the display of the loading indicator screen).

Thanks for mentioning this @minhqdao . It has been already fixed by @henilp105 in this https://github.com/fortran-lang/registry/pull/22/commits/8b167ada37b2d0d79a093eb99e86b5f4f6725adf. Unfortunately, this commit is not in my branch because it was created before this commit was merged. So, I can confirm this issue has already been fixed in `main` branch,

File: registry/src/pages/home.js (hunk 1 above)
Comment (arteevraina) [reply registry/src/pages/home.js:36]: You have already done it here https://github.com/fortran-lang/registry/pull/22/commits/8b167ada37b2d0d79a093eb99e86b5f4f6725adf and that has been merged already. So, I don't think I will have to do it again

**Comment by arteevraina** [comment]
> The enter key now works on the homepage, but once you searched sth and the search bar moves to the top, the enter key stops working.

Thanks for mentioning this @minhqdao . It has been already fixed by @henilp105 in this https://github.com/fortran-lang/registry/pull/22/commits/8b167ada37b2d0d79a093eb99e86b5f4f6725adf. Unfortunately, this commit is not in my branch because it was created before this commit was merged. So, I can confirm this issue has already been fixed in `main` branch,

**Comment by arteevraina** [comment]
> When I start a search from the homepage, there is a loading indicator. But once the search bar moves to the top and I search for packages again, there is none. Seems like some business logic isn't reused here. 🤔

This has been fixed in this commit https://github.com/fortran-lang/registry/pull/23/commits/6cc5275f2af8edf613ea7b5215b3f86756bc5f64.

**Review by perazz — APPROVED** [review]
Looks good to me, thank you @arteevraina.

**Review by henilp105 — APPROVED** [review]
Thanks @arteevraina , looks good to me.

**Pull request merged**